pkg/acsengine/defaults.go

```diff
@@ -233,16 +233,22 @@ func setPropertiesDefaults(cs *api.ContainerService, isUpgrade, isScale bool) (b
 
 	setOrchestratorDefaults(cs, isUpgrade || isScale)
 
-	setMasterNetworkDefaults(properties, isUpgrade)
-
-	setHostedMasterNetworkDefaults(properties)
+	// Set master profile defaults if this cluster configuration includes master node(s)
+	if cs.Properties.MasterProfile != nil {
+		setMasterProfileDefaults(properties, isUpgrade)
+	}
 
-	setAgentNetworkDefaults(properties, isUpgrade, isScale)
+	setAgentProfileDefaults(properties, isUpgrade, isScale)
```

Comment:
Renamed this func because it's not network-specific, it's agent profile-specific.

Reply:
👍 💯